hera_qm/ant_metrics.py

```diff
@@ -431,3 +436,155 @@ def save_antenna_metrics(self, metricsJSONFilename):
 
         with open(metricsJSONFilename, 'w') as outfile:
             json.dump(allMetricsData, outfile, indent=4)
+
+# code for running ant_metrics on a file
+def get_metrics_OptionParser(method_name):
+    """
+    Function to get an OptionParser instance for working with metrics wrappers.
+
+    Args:
+        method_name -- target wrapper, must be "ant_metrics" or "xrfi"
+    Returns:
+        o -- an optparse.OptionParser instance with the relevant options for the selected method
+    """
+    methods = ["ant_metrics", "xrfi"]
+    if method_name not in methods:
+        raise AssertionError('method_name must be one of {}'.format(','.join(methods)))
+
+    o = optparse.OptionParser()
+
+    if method_name == 'ant_metrics':
+        o.set_usage("ant_metrics_run.py -C [calfile] [options] *.uv")
+        aipy.scripting.add_standard_options(o, cal=True)
+        o.add_option('--crossCut', dest='crossCut', default=5, type='float',
+                     help='Modified z-score cut for most cross-polarized antenna. Default 5 "sigmas"')
+        o.add_option('--deadCut', dest='deadCut', default=5, type='float',
+                     help='Modified z-score cut for most likely dead antenna. Default 5 "sigmas"')
+        o.add_option('--extension', dest='extension', default='.ant_metrics.json', type='string',
+                     help='Extension to be appended to the file name. Default is ".ant_metrics.json"')
+        o.add_option('--metrics_path', dest='metrics_path', default='', type='string',
+                     help='Path to save metrics file to. Default is same directory as file')
+    elif method_name == 'xrfi':
+        o.set_usage("xrfi_run.py")
+
+    return o
+
+# helper functions
+def get_pol(fname):
+    """Strips the filename of a HERA visibility to its polarization
+    Args:
+        fname -- name of file (string)
+    Returns:
+        polarization -- polarization label e.g. "xx" (string)
+    """
+    # XXX assumes file naming format
+    # extract just the filename if we're passed a path with periods in it
+    fn = re.findall('zen\.\d{7}\.\d{5}\..*', fname)[0]
+    return fn.split('.')[3]
+
+def generate_fullpol_file_list(files, pol_list):
```

Comment:
This seems like an ugly way to store the files. To my mind, they should be a list of lists...the outer list loops over JD and the inner list loops over polarization. Then you can just iterate over the outer list and get the list you need as an input to `Antenna_Metrics`.